### ISSUE TITLE
llama.cpp

### ISSUE THREAD

**Issue by ctlllll** [FasterDecoding/Medusa#35]

**Comment by kalomaze** [comment]
What type of skills would be needed? Any specific grunt work in mind?

**Comment by kalomaze** [comment]
It looks like a lot of the groundwork is being laid out here:
https://github.com/ggerganov/llama.cpp/pull/3228

**Comment by ctlllll** [comment]
> It looks like a lot of the groundwork is being laid out here with the parallel decoding implementation: [ggerganov/llama.cpp#3228](https://github.com/ggerganov/llama.cpp/pull/3228)

Yeah, that's also what I thought. The tree attention implementation in llama.cpp should be very helpful for integrating Medusa. And I guess the main challenge is the familiarity with their codebase.

**Comment by JianbangZ** [comment]
We are almost done in supporting medusa in llama.cpp. now working on attention mask part and few other things.

**Comment by kalomaze** [comment]
I would like to help with this. Is there any place where I can contact the group behind this and ask questions?

**Comment by ctlllll** [comment]
> I would like to help with finalizing the support for this. Is there any place where I can contact the group behind this project and ask questions?

Hi @kalomaze , that would be great, just sent a friend request on Discord :)

**Comment by kalomaze** [comment]
> > I would like to help with finalizing the support for this. Is there any place where I can contact the group behind this project and ask questions?
> 
> Hi @kalomaze , that would be great, just sent a friend request on Discord :) (I assumed your user name is kalomaze right?)

Sorry, what was your tag? I'm coming back to this now, I figured GitHub would have sent a notification.

**Comment by openconstruct** [comment]
Was this indeed implemented?
